x-pack/platform/plugins/shared/cases/server/connectors/cases/cases_connector_executor.ts

```diff
@@ -819,22 +821,36 @@ export class CasesConnectorExecutor {
       })
       .join(' & ');
 
-    const suffix = `${
-      groupingDescription.length > 0 ? ` - ${GROUPED_BY_TITLE(groupingDescription)}` : ''
-    }${
-      oracleCounter > INITIAL_ORACLE_RECORD_COUNTER ? ` (${oracleCounter})` : ''
-    } (${AUTO_CREATED_TITLE})`;
+    const oracleCounterString =
+      oracleCounter > INITIAL_ORACLE_RECORD_COUNTER ? ` (${oracleCounter})` : '';
 
-    const ruleNameTrimmed = params.rule.name.slice(
-      0,
-      MAX_TITLE_LENGTH - suffix.length - totalDots - 1
-    );
+    const staticSuffixStart = ` - ${GROUPED_BY_TITLE('')}`;
+    const staticSuffixEnd = `${oracleCounterString} (${AUTO_CREATED_TITLE})`;
 
+    const ruleName = params.rule.name;
+    const ruleNameTrimmed = ruleName.slice(0, MAX_RULE_NAME_LENGTH - totalDots);
     const ruleNameTrimmedWithDots =
-      params.rule.name.length > ruleNameTrimmed.length
+      ruleName.length > ruleNameTrimmed.length
         ? `${ruleNameTrimmed}${'.'.repeat(totalDots)}`
         : ruleNameTrimmed;
 
+    const maxSuffixLength =
+      ruleNameTrimmedWithDots.length < MAX_RULE_NAME_LENGTH
+        ? MAX_TITLE_LENGTH - ruleNameTrimmedWithDots.length
+        : MAX_SUFFIX_LENGTH;
+
+    const maxGroupingDescriptionLength =
+      maxSuffixLength - (staticSuffixStart.length + staticSuffixEnd.length);
+
+    const groupingDescriptionTrimmed =
```

Comment:
Could you please add a test for the trimmed grouping description in `x-pack/platform/plugins/shared/cases/server/connectors/cases/cases_connector_executor.test.ts`